src/cider/examples/CiderHashTableExample.cpp

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include <exec/operator/join/HashTableFactory.h>
+#include <gflags/gflags.h>
+#include <algorithm>
+#include <iostream>
+#include <random>
+#include <string>
+#include <vector>
+#include "exec/operator/join/CiderLinearProbingHashTable.h"
+#include "util/Logger.h"
+
+// utils for generate datas
+std::random_device rd;
+std::mt19937 gen(rd());
+
+int random(int low, int high) {
+  std::uniform_int_distribution<> dist(low, high);
+  return dist(gen);
+}
+
+// hashfunction would be in a sepearete hashfunction lib in future, right now is hardcode
+// only for int key
+struct MurmurHash {
```

Comment:
In the end state, do we want to user to specify hash function impl?

Reply:
Hash function should always use different default impls for different data types. I think user should not specify the hashfunction impl unless some people invent some new and great hash function that our lib did not contain.